app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb

```diff
@@ -0,0 +1,150 @@
+            service_id: @authentication_parameters.service_id,
+            account: @authentication_parameters.account,
+            role_name: jwt_identity,
+            constraints: constraints,
+            authentication_request: restriction_validator_class.new(
+              decoded_token: @authentication_parameters.decoded_token
+            )
+          )
+        end
+
+        def validate_and_decode_token
+          @authentication_parameters.decoded_token = validate_and_decode_token_instance.call(
+            authentication_parameters: @authentication_parameters,
+            fetch_signing_key: fetch_signing_key
+          )
+        end
+
+        private
+
+        def jwt_token(authenticator_input)
+          extract_token_from_credentials.call(
+            credentials: authenticator_input.request.body.read
+          )
+        end
+
+        def jwt_identity_from_request
+          @jwt_identity_from_request = identity_provider.jwt_identity
+        end
+
+        def identity_provider
+          @identity_provider = create_identity_provider.call(
+            authentication_parameters: @authentication_parameters
+          )
+        end
+
+        def validate_and_decode_token_instance
+          return @validate_and_decode_token_instance if @validate_and_decode_token_instance
+
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreateValidateAndDecodeTokenInstance.new)
+          @validate_and_decode_token_instance = validate_and_decode_token_class.new(
+            fetch_jwt_claims_to_validate: fetch_jwt_claims_to_validate
+          )
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatedValidateAndDecodeTokenInstance.new)
+          @validate_and_decode_token_instance
+        end
+
+        def fetch_signing_key
+          @fetch_cached_signing_key ||= ::Util::ConcurrencyLimitedCache.new(
+            ::Util::RateLimitedCache.new(
+              fetch_signing_key_interface,
+              refreshes_per_interval: CACHE_REFRESHES_PER_INTERVAL,
+              rate_limit_interval: CACHE_RATE_LIMIT_INTERVAL,
+              logger: @logger
+            ),
+            max_concurrent_requests: CACHE_MAX_CONCURRENT_REQUESTS,
+            logger: @logger
+          )
+        end
+
+        def authentication_parameters_class
+          @authentication_parameters_class ||= Authentication::AuthnJwt::AuthenticationParameters
+        end
+
+        def extract_token_from_credentials
+          @extract_token_from_credentials ||= Authentication::AuthnJwt::InputValidation::ExtractTokenFromCredentials.new
+        end
+
+        def validate_and_decode_token_class
+          @validate_and_decode_token_class ||= Authentication::AuthnJwt::ValidateAndDecode::ValidateAndDecodeToken
+        end
+
+        def validate_resource_restrictions_class
+          @validate_resource_restrictions_class ||= Authentication::ResourceRestrictions::ValidateResourceRestrictions
+        end
```

Comment:
in other non-command classes we set these in the method signature with default values. can we do the same thing here?

Reply:
please do this for all the dependencies in this class